samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManager.java

```diff
@@ -885,28 +938,31 @@ public void stopStores() {
   public void shutdown() {
     // stop all nonsideinputstores including persistent and non-persistent stores
     this.containerModel.getTasks().forEach((taskName, taskModel) ->
-        getNonSideInputStores(taskName).forEach((storeName, store) -> store.stop())
+        getNonSideInputStorageEngines(taskName).forEach((storeName, store) -> store.stop())
     );
 
     this.shouldShutdown = true;
 
     // stop all sideinput consumers and stores
     if (sideInputsPresent()) {
-      sideInputsReadExecutor.shutdownNow();
+      sideInputRunLoop.shutdown();
+      sideInputRunLoopExecutor.shutdownNow();
+
+      this.taskSideInputHandlers.values().forEach(TaskSideInputHandler::stop);
```

Comment:
shouldn't run loop take care of handling shutdown for the tasks within which in turn can shutdown the task side input handlers? why do we need a handle of the side input handlers in CSM?

Reply:
This is actually not the case today; shutdown of tasks is handled by SamzaContainer, not by RunLoop.

We could perhaps change the lifecycle management to place this responsibility on RunLoop instead, but I have not investigated feasibility of that change but rather tried to mimic the current relationship between SamzaContainer, TaskInstance, and RunLoop.